26_CentralLimitBoxBlur/main.cpp

```diff
@@ -0,0 +1,432 @@
+			static_assert( std::is_base_of_v<nbl::asset::IAsset, T> );
+			// The type of the root assets in the bundle is not known until runtime, so this is kinda like a `dynamic_cast` which will return nullptr on type mismatch
+			auto typedAsset = IAsset::castDown<T>( bundle.getContents()[ 0 ] ); // just grab the first asset in the bundle
+			if( !typedAsset )
+			{
+				m_logger->log( "Asset type mismatch want %d got %d !", ILogger::ELL_ERROR, T::AssetType, bundle.getAssetType() );
+
+			}
+			return typedAsset;
+		};
+
+		auto textureToBlur = checkedLoad.operator()< nbl::asset::ICPUImage >( "app_resources/tex.jpg" );
+		if( !textureToBlur )
+		{
+			return logFail( "Failed to load texture!\n" );
+		}
+		const auto& inCpuTexInfo = textureToBlur->getCreationParameters();
+		
+		auto createGPUImages = [ & ](
+			core::bitflag<IGPUImage::E_USAGE_FLAGS> usageFlags,
+			std::string_view name,
+			smart_refctd_ptr<nbl::video::IGPUImage>& imgOut,
+			smart_refctd_ptr<nbl::video::IGPUImageView>& imgViewOut
+		) {
+			video::IGPUImage::SCreationParams gpuImageCreateInfo;
+			gpuImageCreateInfo.flags = inCpuTexInfo.flags;
+			gpuImageCreateInfo.type = inCpuTexInfo.type;
+			gpuImageCreateInfo.extent = inCpuTexInfo.extent;
+			gpuImageCreateInfo.mipLevels = inCpuTexInfo.mipLevels;
+			gpuImageCreateInfo.arrayLayers = inCpuTexInfo.arrayLayers;
+			gpuImageCreateInfo.samples = inCpuTexInfo.samples;
+			gpuImageCreateInfo.tiling = video::IGPUImage::TILING::OPTIMAL;
+			gpuImageCreateInfo.usage = usageFlags | asset::IImage::EUF_TRANSFER_DST_BIT;
+			gpuImageCreateInfo.queueFamilyIndexCount = 0u;
+			gpuImageCreateInfo.queueFamilyIndices = nullptr;
+
+			gpuImageCreateInfo.format = m_physicalDevice->promoteImageFormat(
+				{ inCpuTexInfo.format, gpuImageCreateInfo.usage }, gpuImageCreateInfo.tiling
+			);
+			auto gpuImage = m_device->createImage( std::move( gpuImageCreateInfo ) );
+
+			auto gpuImageMemReqs = gpuImage->getMemoryReqs();
+			gpuImageMemReqs.memoryTypeBits &= m_physicalDevice->getDeviceLocalMemoryTypeBits();
+			m_device->allocate( gpuImageMemReqs, gpuImage.get(), video::IDeviceMemoryAllocation::EMAF_NONE );
+
+			auto imgView = m_device->createImageView( {
+				.flags = IGPUImageView::ECF_NONE,
+				.subUsages = usageFlags,
+				.image = gpuImage,
+				.viewType = IGPUImageView::ET_2D,
+				.format = gpuImageCreateInfo.format
+			} );
+			gpuImage->setObjectDebugName( name.data() );
+			imgView->setObjectDebugName( ( std::string{ name } + "_view" ).c_str() );
+			imgOut = gpuImage;
+			imgViewOut = imgView;
+		};
+
+
+		smart_refctd_ptr<nbl::video::IGPUImage> inputGpuImg;
+		smart_refctd_ptr<nbl::video::IGPUImage> outputGpuImg;
+		smart_refctd_ptr<nbl::video::IGPUImageView> inputGpuImgView;
+		smart_refctd_ptr<nbl::video::IGPUImageView> outputGpuImgView;
+		createGPUImages( IGPUImage::EUF_SAMPLED_BIT, "InputImg", inputGpuImg, inputGpuImgView );
+		createGPUImages( IGPUImage::EUF_STORAGE_BIT, "OutputImg", outputGpuImg, outputGpuImgView );
```

Comment:
btw if you transition to using a swapchain, you'll have to have `maxFramesInFlight` output images and command buffers